src/callsite.jl

```diff
@@ -126,36 +127,53 @@ function Base.print(io::TextWidthLimiter, s::String)
         for c in graphemes(s)
             cwidth = textwidth(c)
             if has_space(io, cwidth)
-                print(io, c)
+                print(io.io, c)
                 io.width += cwidth
             else
                 break
             end
         end
-        print(io, '…')
+        print(io.io, '…')
         io.width += 1
     end
 end
 
+function Base.print(io::TextWidthLimiter, c::Char)
+    tw = textwidth(c)
+    if has_space(io, tw)
+        print(io.io, c)
+        io.width += tw
+        return
+    end
+    return
+end
+
+function Base.take!(io::TextWidthLimiter)
+    io.width = 0
+    return take!(io.io)
+end
+
 function headstring(@nospecialize(T))
     T = widenconst(Base.unwrapva(T))
     if T isa Union || T === Union{}
         return string(T)
     elseif T isa UnionAll
         return headstring(Base.unwrap_unionall(T))
     else
-        return string(T.name)
+        return string(T.name.name)
     end
 end
 
 
 function __show_limited(limiter, name, tt, rt)
+    vastring(@nospecialize(T)) = Base.isvarargtype(T) ? headstring(T)*"..." : string(T)
```

Comment:
I think we will probably need to handle the case where `N` is also specified separately, i.e. something like `::Vararg{T, 3}`.

Reply:
I think(?) those get expanded?

```julia
julia> getidx(A::Array{T}, i::Vararg{Integer,N}) where {T,N} = sin(A[i...])
getidx (generic function with 1 method)
julia> callsites = find_callsites_by_ftt(getidx, Tuple{Matrix{Bool}, Vararg{T,2} where T<:Integer}, optimize=false)
2-element Vector{Cthulhu.Callsite}:
 %2  = call #_apply_iterate(::#iterate{…},::#getindex{…},::Tuple{…},::Tuple{…})
 %3  = call sin(::Bool)::Float64

julia> cs = callsites[1]
%2  = call #_apply_iterate(::typeof(iterate),::typeof(getindex),::Tuple{Matrix{Bool}},::Tuple{Integer, Integer})::Bool
```

Reply:
@simeonschaub, if you still think this needs doing I'm happy to do it in a future PR.